src/c-dvar-reader.c

```diff
@@ -436,6 +436,21 @@ static int c_dvar_ff(CDVar *var) {
         do {
                 if (var->current->n_type && (var->current->container != 'a' || c_dvar_more(var))) {
                         c = var->current->i_type->element;
+                        if (var->current->container == 'a'
+                                && var->current->i_type->size > 0
+                                && var->current->i_type->basic
+                                && c != 'b') {
```

Comment:
This will skip past an entire array even if you call `c_dvar_skip("*")` while already in an array. This needs to check for `depth > 0`, I think.